### DISCUSSION TITLE
Fretish/CoCosim Simulation Time Handling

### DISCUSSION THREAD

**Discussion by StefanoCentomo** [NASA-SW-VnV/fret#59] [Q&A]
Hello, 

How can I manage time granularity from Fret to Cocosim?

For instance, let consider this simple requirement:
> if airspeed < 100 Aircraft shall within 10 seconds satisfy airspeed > 100

I have imported that requirement into CoCosim and I noticed that the the contract block does not have any timing settings in its properties (the usual sample rate for simulink block).

Testing this block  I've seen that the time is managed with a sample time of 1 second, but if I would change the "sample rate" of the monitor (like for instance 2 seconds instead of 1) is it possible to do that?
If I change the sample rate of Contracts block I've noticed that the internal tick  does not change its time knowledge. 

is there any suggestion to externally set the internal  simulated time of  the monitor?

This will  enable variable-step solvers from Simulink (i.e. without forcing monitor ports sample rate), 
Are you planning to add internal timing storage to the monitor?

Best,
Stefano

**Comment by andreaskatis** [comment]
Hi Stefano,

This is a very interesting question and I would like to think and understand more about it. In the meantime, since this is a discussion that is more relevant to CoCoSim than FRET, would you mind creating a new post there?

https://github.com/NASA-SW-VnV/CoCoSim/discussions

I would also appreciate it if you could provide a minimal Simulink example that would demonstrate what it is you want to achieve.

Best Regards,

Andreas

**Comment by StefanoCentomo** [comment]
Hi Andreas,

Ok. 
I'll open a new discussion in Cocosim, with the simulink example.


Best,
Stefano
